temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -2157,17 +2230,15 @@ private void timeoutNexusOperation(
     }
   }
 
-  private void addNexusOperationRetryTimer(
-      RequestContext ctx, StateMachine<NexusOperationData> operation) {
+  private void retryNexusTask(RequestContext ctx, StateMachine<NexusOperationData> operation) {
+    State prevState = operation.getState();
     NexusOperationData data = operation.getData();
     int attempt = data.getAttempt();
-    Duration nextDelay = ProtobufTimeUtils.toJavaDuration(data.nextBackoffInterval);
-    data.nextAttemptScheduleTime = clock.getAsLong() + nextDelay.toMillis();
     ctx.addTimer(
```

Comment:
I see there's a TODO below, can you address it?

Reply:
Also not seeing where the request timeout is enforced on the retry here (mentioned elsewhere).